crates/ott-balancer-bin/src/balancer.rs

```diff
@@ -622,17 +622,23 @@ pub async fn dispatch_monolith_message(
                     };
 
                     // TODO: also handle the case where the client_id is Some
-
-                    // broadcast to all clients
-                    debug!("broadcasting to clients in room: {:?}", room.name());
-                    // TODO: optimize this using a broadcast channel
                     let built_msg = Message::text(msg.payload.to_string());
-                    for client in room.clients() {
-                        let Some(client) = ctx_read.clients.get(client) else {
-                            anyhow::bail!("client not found");
-                        };
-
-                        client.send(built_msg.clone()).await?;
+                    
+                    match ctx_read.clients.get(&msg.client_id) {
```

Comment:
This needs to check if `msg.client_id` is `Some`. That way we don't accidentally broadcast a message that was only intended for one client.